gax-grpc/src/main/java/com/google/api/gax/grpc/UnaryCallable.java

```diff
@@ -328,4 +329,8 @@ public ResponseT call(RequestT request) {
     return new UnaryCallable<>(
         new BatchingCallable<>(callable, batchingDescriptor, batcherFactory), channel, settings);
   }
+
+  UnaryCallable<RequestT, ResponseT> auth(CallCredentials credentials) {
```

Comment:
I don't think `UnaryCallable` should use the provider - it doesn't use the provider for channel or executor, so it would be inconsistent.

Maybe we want a `NoCredentialsProvider` which would not set any credentials, similar to `NoCredentials` in google-cloud-java.